bigtable/tests/unit/test_table.py

```diff
@@ -411,6 +411,98 @@ def _list_column_families_helper(self):
     def test_list_column_families(self):
         self._list_column_families_helper()
 
+    def test_get_cluster_states(self):
+        from google.cloud.bigtable_admin_v2.gapic import (
+            bigtable_table_admin_client)
+        from google.cloud.bigtable.enums import Table as enum_table
+        from google.cloud.bigtable.table import _ClusterState
+        INITIALIZING = enum_table.ReplicationState.INITIALIZING
+        PLANNED_MAINTENANCE = enum_table.ReplicationState.PLANNED_MAINTENANCE
+        READY = enum_table.ReplicationState.READY
+
+        table_api = bigtable_table_admin_client.BigtableTableAdminClient(
+            mock.Mock())
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+        instance = client.instance(instance_id=self.INSTANCE_ID)
+        table = self._make_one(self.TABLE_ID, instance)
+
+        response_pb = _TablePB(
+            cluster_states={'cluster-id1': _ClusterStatePB(INITIALIZING),
+                            'cluster-id2': _ClusterStatePB(
+                                PLANNED_MAINTENANCE),
+                            'cluster-id3': _ClusterStatePB(READY),
+                            },
+        )
+
+        # Patch the stub used by the API method.
+        client._table_admin_client = table_api
+        bigtable_table_stub = (
+            client._table_admin_client.bigtable_table_admin_stub)
+        bigtable_table_stub.GetTable.side_effect = [response_pb]
+
+        # build expected result
+        expected_result = {
+            u'cluster-id1': _ClusterState(INITIALIZING),
+            u'cluster-id2': _ClusterState(PLANNED_MAINTENANCE),
+            u'cluster-id3': _ClusterState(READY)
+        }
+
+        # Perform the method and check the result.
+        result = table.get_cluster_states()
+        self.assertEqual(result, expected_result)
+
+        # test for __eq__ and __ne__
+        self.assertNotEqual(_ClusterState(READY), enum_table())
+        self.assertNotEqual(_ClusterState(READY),
+                            _ClusterState(INITIALIZING))
```

Comment:
The tests above are duplicates of the tests lower down lines [502-504](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5790/files#diff-d495c61ac7ca28885a9cb09130c527c1R502). I think it will be better to make separate tests for `__eq__` and `__ne__` of class `_ClusterState`, similar to the same tests for `Table` class [here](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/bigtable/tests/unit/test_table.py#L225-L241) and [here](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/bigtable/tests/unit/test_table.py#L243-L256)

You could add a class _Other at the end of the file
```python
class _Other(object):

    def __init__(self, replication_state):
        self.replication_state = replication_state

```
and then add following tests instead of the asserts above and on lines 501-504.
```python
    def test___eq___ClusterState(self):
        from google.cloud.bigtable.table import _ClusterState
        from google.cloud.bigtable.enums import Table
        ready = Table.ReplicationState.READY
        cluster_state1 = _ClusterState(ready)
        cluster_state2 = _ClusterState(ready)
        self.assertTrue(cluster_state1 == cluster_state2)

    def test___eq___ClusterState_type_differ(self):
        from google.cloud.bigtable.table import _ClusterState
        from google.cloud.bigtable.enums import Table
        ready = Table.ReplicationState.READY
        cluster_state1 = _ClusterState(ready)
        other = _Other(ready)
        self.assertFalse(cluster_state1 == other)

    def test___ne___ClusterState_same_value(self):
        from google.cloud.bigtable.table import _ClusterState
        from google.cloud.bigtable.enums import Table
        ready = Table.ReplicationState.READY
        cluster_state1 = _ClusterState(ready)
        cluster_state2 = _ClusterState(ready)
        self.assertFalse(cluster_state1 != cluster_state2)

    def test___ne___ClusterState(self):
        from google.cloud.bigtable.table import _ClusterState
        from google.cloud.bigtable.enums import Table
        ready = Table.ReplicationState.READY
        initializing = Table.ReplicationState.INITIALIZING
        cluster_state1 = _ClusterState(ready)
        cluster_state2 = _ClusterState(initializing)
        self.assertTrue(cluster_state1 != cluster_state2)

```
This will make make code coverage happy as well